wbobeirne/joule-extension
# Fix Firefox CORS & Lightning App onboarding button

Closes #99. Relies on #148 to fix lightning-app crashing Joule.

### Description

This is two-parter. The first is that it fixes the CORS issues outlined in #99. As noted in [the bug report I opened with Mozilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1514963), Firefox will allow CORS bypass if you leave off the port.

The second thing done here is to add a distinct option to onboarding for the Lightning App. This will check port 8086 specifically, and show different paths for where Macaroons might be.

### Steps to Test

1. Run the lightning-app application, and get your node fully setup & synced.
2. Go through onboarding in Firefox, and select the lightning app option. Confirm you're able to get the URL working by fixing SSL issues.
3. Confirm the macaroon paths are correct for your operating system, and that uploading them works.
4. Finish onboarding, confirm endpoints work without CORS issues.
5. Go through onboarding again as a remote / local node, confirm macaroon paths are correct.

## Screenshots

### New option

<img width="400" alt="screen shot 2019-01-18 at 4 21 02 pm" src="https://user-images.githubusercontent.com/649992/51413517-4076e700-1b3d-11e9-881c-18d659c4b6f0.png">

### Correct paths

<img width="400" alt="screen shot 2019-01-18 at 4 21 13 pm" src="https://user-images.githubusercontent.com/649992/51413520-4240aa80-1b3d-11e9-80cb-f092f9d60c71.png">
